src/spikeinterface/sorters/external/kilosortbase.py

```diff
@@ -53,21 +60,33 @@ def _generate_channel_map_file(recording, sorter_output_folder):
         ycoords = ([p[1] for p in positions],)
         kcoords = (groups,)
 
-        channel_map = {}
-        channel_map["Nchannels"] = nchan
-        channel_map["connected"] = np.full((nchan, 1), True)
-        channel_map["chanMap0ind"] = np.arange(nchan)
-        channel_map["chanMap"] = channel_map["chanMap0ind"] + 1
-
-        channel_map["xcoords"] = np.array(xcoords).astype(float)
-        channel_map["ycoords"] = np.array(ycoords).astype(float)
-        channel_map["kcoords"] = np.array(kcoords).astype(float)
-
-        sample_rate = recording.get_sampling_frequency()
-        channel_map["fs"] = float(sample_rate)
-        import scipy.io
-
-        scipy.io.savemat(str(sorter_output_folder / "chanMap.mat"), channel_map)
+        if format == "mat":
+            channel_map = {}
+            channel_map["Nchannels"] = nchan
+            channel_map["connected"] = np.full((nchan, 1), True)
+            channel_map["chanMap0ind"] = np.arange(nchan)
+            channel_map["chanMap"] = channel_map["chanMap0ind"] + 1
+
+            channel_map["xcoords"] = np.array(xcoords).astype(float)
+            channel_map["ycoords"] = np.array(ycoords).astype(float)
+            channel_map["kcoords"] = np.array(kcoords).astype(float)
+
+            sample_rate = recording.get_sampling_frequency()
+            channel_map["fs"] = float(sample_rate)
+            import scipy.io
+
+            scipy.io.savemat(str(sorter_output_folder / "chanMap.mat"), channel_map)
+        elif format == "json":
+            from kilosort.io import save_probe
+
+            probe = {
+                "chanMap": np.arange(nchan),
+                "xc": np.array(xcoords[0]).astype(float).squeeze(),
+                "yc": np.array(ycoords[0]).astype(float).squeeze(),
+                "kcoords": np.array(kcoords).astype(float).squeeze(),
+                "n_chan": nchan,
+            }
+            save_probe(probe, str(sorter_output_folder / "chanMap.json"))
```

Comment:
Since this is KS4 specific, I would remove it from here and leave it in the `setup_recording` of the KS4 wrapper